### PR TITLE
fix(pat relateditems): Import Select2 module before calling it when not yet initialized

### PR DESCRIPTION
**Pull request by frapell**
This PR fixes gh-1341

### CODE REVIEW

**Comment by frapell** [comment]
@thet Mind taking a look at this if you have some time? My understanding is that the `new` operator doesn't return a promise, so it cannot be awaited... My first approach at fixing this issue was to add an `await` in https://github.com/plone/mockup/blob/c9d330a8e06aa0b4e193a9497f3eb5558efc2811/src/pat/tinymce/js/links.js#L101 but that doesn't work and https://github.com/plone/mockup/blob/c9d330a8e06aa0b4e193a9497f3eb5558efc2811/src/pat/tinymce/js/links.js#L109 gets called with the `select2` module not yet available.

I don't like my solution very much, but I believe a proper solution would require extensive refactoring, and could potentially break in unexpected places...

Thoughts?

File: src/pat/relateditems/relateditems.js [hunk 1]
```diff
@@ -456,7 +456,8 @@ export default Base.extend({
         this.emit("after-browse");
     },
 
-    selectItem(item) {
+    async selectItem(item) {
+        await import("select2");
```
Comment (thet) [inline src/pat/relateditems/relateditems.js:460]: LGTM

**Comment by thet** [comment]
Hi, I think using the `await_pattern_init` from Patternslib should work.
Import:
```
import events from "@patternslib/patternslib/src/core/events";
```

then:
```
createRelatedItems: async function () {
    [...]
    this.relatedItems = new RelatedItems(this.getEl(), options);
    await events.await_pattern_init(this.relatedItems)
}
```

There should be many examples in Patternslib.

Hope that helps!

I'm happy to review this later - today it's bedtime.

**Comment by frapell** [comment]
@thet Thank you for that! had no idea about the `await_pattern_init` and I think it is a better solution. I force pushed the commit replacing the previous one to avoid excessive commits

**Comment by frapell** [comment]
@thet I included my latest commit into this PR, which is totally unrelated to this fix, but I think it is a nice to have. Right now, it is not possible to add the "Insert Link" or "Insert Image" to the menu, they can only be added as buttons to the toolbar.
I added calls to `addMenuItem` for them (https://www.tiny.cloud/docs/tinymce/6/apis/tinymce.editor.ui.registry/#addMenuItem) and now you can add those actions to the menu:
image

Let me know if you want me to revert it and submit a separate PR for that

**Review by thet — APPROVED** [review]
LGTM.

regarding the add image/add link in the insert menu - totally fine for me.
But you have this commit message: "Include ability to add the insert image and insert link in the insert menu". Shouldn't it be: "Include the insert image and insert link in the insert menu"? Because you aren't adding a ability but the menu items itself, right?

**Comment by frapell** [comment]
@thet No, you still need to add `ploneimage` and `plonelink` to the JSON config for the menu in `/@@tinymce-controlpanel`. What happens today without this code change is that if you cannot have the buttons showing up in the menu, even adding those entries to the config.

**Pull request merged**